### PR TITLE
[nhk] Add extractor for VoD.

### PR DESCRIPTION
**Pull request by Xender**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extractor for NHK VoD.
Was requested by at least 2 people in #4437 and by myself.

Heavily based on Vice extractor, which also sometimes uses Ooyala player.

I haven't find a suitable test case, as all videos I've visited are available only for a limited period of time.

Example URL: http://www3.nhk.or.jp/nhkworld/en/vod/tokyofashion/20160815.html

This is my first contribution to youtube-dl. Please point me out if I missed something important.

### CODE REVIEW

File: youtube_dl/extractor/nhk.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class NhkVodIE(InfoExtractor):
+    _VALID_URL = r'http://www3.nhk.or.jp/nhkworld/en/vod/(?P<id>.+)\.html'
+    # _TESTS = []  # TODO videos I've found are all available only for a limited period of time. That automatically makes such test cases obsolete. :/
```
Comment (TRox1972) [inline youtube_dl/extractor/nhk.py]: If you can't find any permanent videos, add a temporary video and `'skip_download'` to `True`. This at least tests the `_VALID_URL`.

**Comment by dstftw** [comment]
Provide [an example URL](https://github.com/rg3/youtube-dl/blob/master/README.md#is-the-description-of-the-issue-itself-sufficient).

File: youtube_dl/extractor/nhk.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        embed_code = self._search_regex(
+            r"nw_vod_ooplayer\('movie-area', '([^']+)'\);",
```
Comment (TRox1972) [inline youtube_dl/extractor/nhk.py]: Single quotes consistency

File: youtube_dl/extractor/nhk.py (hunk 1 above)
Comment (Xender) [inline youtube_dl/extractor/nhk.py]: Fixed in later commit. The test downloads the metadata, though, so it will become outdated anyway.

**Comment by Xender** [comment]
@dstftw - Edited first post. Will 404 in a few days anyway, though.

File: youtube_dl/extractor/nhk.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            'skip_download': True  # Videos available only for a limited period of time.
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        embed_code = self._search_regex(
+            r'''nw_vod_ooplayer\('movie-area', '([^']+)'\);''',
```
Comment (dstftw) [inline youtube_dl/extractor/nhk.py]: Usage of single quotes only makes sense when string itself does not contain single quotes.

File: youtube_dl/extractor/nhk.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class NhkVodIE(InfoExtractor):
+    _VALID_URL = r'http://www3.nhk.or.jp/nhkworld/en/vod/(?P<id>.+)\.html'
```
Comment (dstftw) [inline youtube_dl/extractor/nhk.py]: Dots must be escaped.

**Comment by Xender** [comment]
I've tried to fix the mentioned issues and squashed the whole thing back into single commit as it started getting too much commits for something which is an initial request for a new feature/extractor.

The unsquashed version is on [Xender:nhk-vod-unsquashed](https://github.com/Xender/youtube-dl/tree/nhk-vod-unsquashed), but I quess it'd be unnecessary clutter to merge that.

File: youtube_dl/extractor/nhk.py (hunk 3 above)
Comment (Xender) [inline youtube_dl/extractor/nhk.py]: The original version had double quotes, which was commented on by @TRox1972 (I received the notification email from Github). I think he removed the comment later, though.

After all I think I prefer the triple single quotes over double quotes here anyway, as they more ambiguously indicate that the string contains special characters (be it line break or normal delimiter).

Would you consider it acceptable or do I have to change it back to double quotes or single quotes with every inner single quote escaped (later being IMHO more disruptive to read)?

**Pull request closed without merging**